Sources/Core/Extensions/Other/XCTestCase+URLRequest.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  XCTestCase+URLRequest.swift
+//  PovioKit
+//
+//  Created by Gentian Barileva on 09/09/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import XCTest
+
+public extension XCTestCase {
+  func XCTAssertEqualURLRequests(_ lhs: URLRequest?, _ rhs: URLRequest?, file: StaticString = #filePath, line: UInt = #line) {
+    XCTAssertEqual(lhs, rhs, file: file, line: line)
+    guard let lhs = lhs, let rhs = rhs else { return }
+    XCTAssertEqual(lhs.url, rhs.url, "url", file: file, line: line)
+    XCTAssertEqual(lhs.httpMethod, rhs.httpMethod, "httpMethod", file: file, line: line)
+    XCTAssertEqual(lhs.httpBody ?? Data(), rhs.httpBody ?? Data(), "httpBody", file: file, line: line)
+    XCTAssertEqual(lhs.allHTTPHeaderFields ?? [:], rhs.allHTTPHeaderFields ?? [:], "allHTTPHeaderFields", file: file, line: line)  }
```

Comment:
You can omit the default values (`Data()` and `[:]`).

Reply:
The reason why I defaulted to Empty Data and Dictionary is that it happens that URLRequest sets nil values as empty, and this can mess up with testing.
<img width="637" alt="Screen Shot 2022-09-12 at 2 44 22 PM" src="https://user-images.githubusercontent.com/15278390/189656924-ee188a4a-1998-42a5-8e71-8e4ceff19682.png">